File: solutions/go/rna-transcription/1/rna_transcription.go
package strand

func ToRNA(dna string) string {
	s := ""
    for _,x := range dna{
        switch x{
            case 'G':
            	s += string('C')
            case 'C':
            	s += string('G')
            case 'T':
            	s += string('A')
            case 'A':
            	s += string('U')
            default:
            	return ""
        }
    }
    return s
}

File: solutions/c/rna-transcription/1/rna_transcription.c
#include "rna_transcription.h"

char *to_rna(const char *dna){
    char *answer = malloc(strlen(dna)+1);
    int i = 0;
    while (*dna!='\0'){
        switch (*dna){
            case 'G':
                answer[i] = 'C';
                break;
            case 'C':
                answer[i] = 'G';
                break;
            case 'T':
                answer[i] = 'A';
                break;
            case 'A':
                answer[i] = 'U';
                break;
        }
        i++;
        dna++;
    }
    answer[i] = '\0';
    return answer;
}